### ISSUE TITLE
ps0 will be OOM using MonitoredTrainingSession when workers too many

### ISSUE THREAD

**Issue by jackonan** [tensorflow/tensorflow#22137]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: No
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: 1.4
- **Python version**: 2.7.5
- **Bazel version (if compiling from source)**: 1.9
- **GCC/Compiler version (if compiling from source)**: 4.9.2
- **CUDA/cuDNN version**: CPU
- **GPU model and memory**: CPU
- **Exact command to reproduce**: Using MonitoredTrainingSession and thousands of workers.

### Describe the problem
My model is large and complex. Thousands of worker&ps are needed and MonitoredTrainingSession is used. In the stage of initialization, the memory of ps0 will raise up to 100G and then OOM, but other ps work correctly. Op device placement is tf.train.replica_device_setter and no other special strategy.

I open the log_device_placement and found that all ops related with report_uninitialized_xxx are placed on ps0. I think this is the root cause.

I change the code as PR #22136 , and then it works correctly.

**Comment by jackonan** [comment]
@drpngx Hi, do you have some updates?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ispirmustafa: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @karmel: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @martinwicke, @karmel: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tilakrayal** [comment]
@jackonan,
Respected PR [#22136](https://github.com/tensorflow/tensorflow/pull/22136) was closed. Could you please take a look at the PR and confirm if the issue still persists. Thank you!
